js/5etools-template.js

```diff
@@ -0,0 +1,502 @@
+const d20plusTemplate = function () {
+	d20plus.template5e = {};
+
+	d20plus.template5e.addCustomHTML = function () {
+		// Object to get data urls because they get set after the categories object is created
+		const dataUrls = {
+			"spell": spellDataUrls,
+			"monster": monsterDataUrls,
+			"class": classDataUrls,
+		}
+
+		function populateDropdown (dropdownId, inputFieldId, baseUrl, srcUrlObject, defaultSel, brewProps) {
+			const defaultUrl = defaultSel ? d20plus.formSrcUrl(baseUrl, srcUrlObject[defaultSel]) : "";
+			$(inputFieldId).val(defaultUrl);
+			const dropdown = $(dropdownId);
+			$.each(Object.keys(srcUrlObject), function (i, src) {
+				dropdown.append($("<option>", {
+					value: d20plus.formSrcUrl(baseUrl, srcUrlObject[src]),
+					text: brewProps.includes("class") ? src.uppercaseFirst() : Parser.sourceJsonToFullCompactPrefix(src),
+				}));
+			});
+			dropdown.append($("<option>", {
+				value: "",
+				text: "Custom",
+			}));
+
+			const dataList = [];
+			const seenPaths = new Set();
+			brewProps.forEach(prop => {
+				Object.entries(brewIndex[prop] || {})
+					.forEach(([path, dir]) => {
+						if (seenPaths.has(path)) return;
+						seenPaths.add(path);
+						dataList.push({
+							download_url: DataUtil.brew.getFileUrl(path),
+							path,
+							name: path.split("/").slice(1).join("/"),
+						});
+					});
+			});
+			dataList.sort((a, b) => SortUtil.ascSortLower(a.name, b.name)).forEach(it => {
+				dropdown.append($("<option>", {
+					value: `${it.download_url}${d20plus.ut.getAntiCacheSuffix()}`,
+					text: `Homebrew: ${it.name.trim().replace(/\.json$/i, "")}`,
+				}));
+			});
+
+			dropdown.val(defaultUrl);
+			dropdown.change(function () {
+				$(inputFieldId).val(this.value);
+			});
+		}
+
+		function populateBasicDropdown (dropdownId, inputFieldId, defaultSel, brewProps, addForPlayers) {
```

Comment:
as above; `d20plus.template5e._populateBasicDropdown` (probably just yank out the inner `doPopulate`)

Reply:
Because doPopulate is called twice, I'm going to be lazy and keep it unless you care a lot.